test/transcoding/spirv-types.ll

```diff
@@ -1,12 +1,13 @@
 ;; Test SPIR-V opaque types
 ;;
 ; RUN: llvm-as %s -o %t.bc
-; RUN: llvm-spirv %t.bc -spirv-text -o %t.spv.txt
+; RUN: llvm-spirv -preserve-ocl-kernel-arg-type-metadata-through-string %t.bc -spirv-text -o %t.spv.txt
```

Comment:
Likewise, these should be renamed, or extra RUN lines and CHECK prefixes added.

Reply:
I'll go through the tests and make the appropriate changes in RUN and CHECK strings. But I'm a bit uncertain about "non-standard" functionality definition. I acknowledge, that the way of how these two metadata are being translated is a workaround, but the metadata themselves are defined in opencl-spir-12, so it'd better to be translated somehow. Do you think, that it's better at some point to introduce a SPIR-V extension to have their representation?

Reply:
Yes, a SPIR-V extension should have been introduced in the first place for this. Please don't forget to follow up by adding a SPIR-V extension.